server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreIT.java

```diff
@@ -212,17 +224,55 @@ private void testRestoreFlowMultipleIndices(int numberOfIterations, boolean invo
         ensureRed(indices);
         internalCluster().startDataOnlyNodes(3);
 
-        assertAcked(client().admin().indices().prepareClose(indices));
+        boolean restoreAllShards = randomBoolean();
+        if (restoreAllShards) {
+            assertAcked(client().admin().indices().prepareClose(indices));
+        }
         client().admin()
             .cluster()
-            .restoreRemoteStore(new RestoreRemoteStoreRequest().indices(INDEX_NAMES_WILDCARD.split(",")), PlainActionFuture.newFuture());
+            .restoreRemoteStore(
+                new RestoreRemoteStoreRequest().indices(INDEX_NAMES_WILDCARD.split(",")).restoreAllShards(restoreAllShards),
+                PlainActionFuture.newFuture()
+            );
         ensureGreen(indices);
         for (String index : indices) {
             assertEquals(shardCount, getNumShards(index).totalNumShards);
             verifyRestoredData(indicesStats.get(index), true, index);
         }
     }
 
+    public void testRestoreFlowAllShardsNoRedIndex() throws InterruptedException {
+        int shardCount = randomIntBetween(1, 5);
+        prepareCluster(0, 3, INDEX_NAME, 0, shardCount);
+        assertEquals(shardCount, getNumShards(INDEX_NAME).totalNumShards);
+
+        PlainActionFuture<RestoreRemoteStoreResponse> future = PlainActionFuture.newFuture();
+        client().admin().cluster().restoreRemoteStore(new RestoreRemoteStoreRequest().indices(INDEX_NAME).restoreAllShards(true), future);
+        try {
+            future.get();
+        } catch (ExecutionException e) {
+            // If the request goes to co-ordinator, e.getCause() can be RemoteTransportException
+            assertTrue(e.getCause() instanceof IllegalStateException || e.getCause().getCause() instanceof IllegalStateException);
+        }
+    }
```

Comment:
> Does this mean that while triggering a restore with restoreAllShards as true on a red index without first closing the index will throw an exception, doing the same on a yellow/green index without closing the index first will be a no-op and we expect no exception even if this would otherwise-violate what the API expects to be a valid request?

Exception will be thrown if we trigger restore for an index in any state (green, yellow, red) without closing the index first.

> Also, is this test targeted for an empty index or do we need to add a step to index some docs?

No, forgot to add docs. Will add.

> Do we need any assertions post the restore?

At least for this test, restore will not go through. Do you mean to check if the index is in the same state as before?

Reply:
Thanks Sachin

> At least for this test, restore will not go through. Do you mean to check if the index is in the same state as before?

My bad, no restore makes sense here. I do think asserting that the index is the same as before would be a good idea (can be abstracted out into a common helper maybe, which can be reused across remote store tests). It could help us catch regressions on any unwanted index modifications for cases like this one.